app/lib/clear_partner_related_data.rb

```diff
@@ -35,4 +36,12 @@ def clear_partner_name
     @storage.save_form(form)
   end
 
+  def clear_partner_ni
+    form = Forms::PartnerNationalInsurance.new
+    form_attributes = @storage.load_form(form)
+    form_attributes.delete('partner_ni_number_present')
+    form_attributes.delete('number')
+    form = Forms::PartnerNationalInsurance.new(form_attributes)
```

Comment:
You are shadowing 'form' here - it is defined on line 40 - is this intentional ?  It reads a bit confusing